common/autoware_universe_utils/src/geometry/ear_clipping.cpp

```diff
@@ -20,311 +20,392 @@
 namespace autoware::universe_utils
 {
 
-void EarClipping::operator()(const Polygon2d & polygon)
+std::size_t construct_point(std::size_t index, const Point2d & point, std::vector<Point> & points)
 {
-  indices.clear();
-  vertices = 0;
-
-  if (polygon.outer().size() == 0) return;
-
-  std::size_t len = 0;
+  points.emplace_back(index, point);
+  return index;
```

Comment:
I think the `index` is not necessary. If so, please remove it, use `push_back(point)` instead of `emplace_back`, and set the return to either void, or `points.size() - 1`.